packages/ai-mcp/src/browser/mcp-frontend-application-contribution.ts

```diff
@@ -72,8 +75,9 @@ export class McpFrontendApplicationContribution implements FrontendApplicationCo
                 MCP_SERVERS_PREF,
                 {}
             ));
-            this.syncServers(servers);
             this.prevServers = this.convertToMap(servers);
+            this.syncServers(this.prevServers);
+            this.autoStartServers(this.prevServers);
```

Comment:
Some usability questions:
- This will only auto start servers when the frontend is started. Should we also auto start servers which are added to the preferences with `autostart: true` or when their `autostart` value is added or changed?
- If a frontend is started with a server set to `autostart: false` (or non existing), should we then stop the server in case it is already running?

Reply:
1.: I decided to not start them on preference change (this is documented in the settings). Reasoning was that if you start a server manually, you get the message with the functions. If you set up a new server, you probably want to start it manually once. If you have a strong feeling to change this behavior, its fine, but I found it a bit more logical this way.
2.: I decided no, because this would break other frontends that still use this server. Also, "removing auto start", does not imply "stop it" to me. You can still manually stop them, though.